### PR TITLE
Add documentation on authentication

### PR DESCRIPTION
**Pull request by flaiker**
Something like this?

Haven't actually tried building the page as WSL is particularly moody today.

### CODE REVIEW

**Comment by larshp** [comment]
thanks, no problems in build,

image

**Comment by FreHu** [comment]
Maybe mention that if an access token is used, it goes in the password field. I've seen a few people surprised by this.

File: docs/guide-authentication.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+![Authentication dialog](img/auth_dialog.png)
+
+## Two factor authentication
+
+Most git hosts allow for two factor authentication. Relevant for abapGit is how to authenticate with 2FA enabled on the command line when using git over HTTP. This is usually done using a token which can be used instead of the normal password. Check the documentation of your git host on how to this in detail ([GitHub](https://docs.github.com/en/github/authenticating-to-github/accessing-github-using-two-factor-authentication#using-two-factor-authentication-with-the-command-line), [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html), [Bitbucket](https://support.atlassian.com/bitbucket-cloud/docs/app-passwords/), [Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate)). Note that the username might be different from your normal one when using the token as the password.
```
Comment (mbtools) [inline docs/guide-authentication.md:13]: on how to **do** this

File: docs/guide-authentication.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+To store passwords or tokens you might want to use a password manager. To enable autofill you can match the window title of the entry to the following pattern:
+
+`NPL(*)/001 Login: github.com`
+
+Note the part `NPL(*)/001` only appears if you have `Show system name in taskbar button` enabled in the SAP GUI options.
+
+## Using a custom authentication mechanism
+
+You can use the exit [`CREATE_HTTP_CLIENT`](ref-exits.html#create_http_client) to implement a different authentication yourself by returning an already authenticated http client instance. See the exit documentation for examples.
```
Comment (mbtools) [inline docs/guide-authentication.md:25]: add the link to "exit documentation" (it's kind of hidden behind the technical name)

File: docs/guide-authentication.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+title: Authentication
+category: online projects
+order: 70
+---
+
+Online projects may require authentication to create branches or commits on the remote or even fetch the content in case of private repositories. Since abapGit uses git over HTTP the authentication mechanism is Basic Authentication (RFC 7617) using username and password. If the remote requires authentication for an action a popup will appear to enable the user to provide the necessary data. Once authentication is successful the session remains intact until abapGit is restarted.
+
```
Comment (mbtools) [inline docs/guide-authentication.md:8]: i think HTTPS is mandatory. add link to the "SSL setup" which is a prerequisite

**Comment by mbtools** [comment]
Very helpful. https://github.com/larshp/abapGit/pull/3934 should help a bit, too

File: docs/guide-authentication.md (hunk 3 above)
Comment (flaiker) [reply docs/guide-authentication.md:8]: No, it's not.

**Comment by larshp** [comment]
thanks 👍

**Pull request merged**